clients/da-vinci-client/src/main/java/com/linkedin/davinci/blob/BlobSnapshotManager.java

```diff
@@ -0,0 +1,131 @@
+package com.linkedin.davinci.blob;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.store.rocksdb.RocksDBUtils;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import java.io.File;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.rocksdb.Checkpoint;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+
+
+public class BlobSnapshotManager {
+  Map<String, Map<Integer, AtomicLong>> concurrentSnapshotUsers;
+  Map<String, Map<Integer, Long>> snapShotTimestamps;
+  long snapshotRetentionTime;
+  private final String basePath;
+
+  @VisibleForTesting
+  protected Checkpoint createCheckpoint(RocksDB rocksDB) {
+    return Checkpoint.create(rocksDB);
+  }
+
+  private static final Logger LOGGER = LogManager.getLogger(BlobSnapshotManager.class);
+
+  /**
+   * Constructor for the BlobSnapshotManager
+   */
+  public BlobSnapshotManager(String basePath, long snapshotRetentionTime) {
```

Comment:
I think we need to make a config in ConfigKeys, but maybe we can leave it for now and do it later when we are integrating this logic into the BlobTransferManager class.